sql/src/main/java/io/crate/protocols/postgres/types/PGGeoPointType.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres.types;
+
+import com.google.common.primitives.Bytes;
+import org.jboss.netty.buffer.ChannelBuffer;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+class PGGeoPointType extends PGType {
+
+    private static final int OID = 600;
+    private static final int TYPE_LEN = 16;
+    private static final int TYPE_MOD = -1;
+
+    public static final PGGeoPointType INSTANCE = new PGGeoPointType();
+
+    private PGGeoPointType() {
+        super(OID, TYPE_LEN, TYPE_MOD, "point");
+    }
+
+    @Override
+    public int writeAsBinary(ChannelBuffer buffer, @Nonnull Object value) {
+        buffer.writeInt(TYPE_LEN);
+        buffer.writeDouble(((double[]) value)[0]);
+        buffer.writeDouble(((double[]) value)[1]);
+        return INT32_BYTE_SIZE + TYPE_LEN;
+    }
+
+    @Override
+    public Object readBinaryValue(ChannelBuffer buffer, int valueLength) {
+        assert valueLength == TYPE_LEN : "length should be " + TYPE_LEN + " because point is 2*64 bit double. " +
+                                         "Actual length: " + valueLength;
+        return new Double[] {buffer.readDouble(), buffer.readDouble()};
+    }
+
+    @Override
+    byte[] encodeAsUTF8Text(@Nonnull Object array) {
```

Comment:
would it make sense to create static variants of the Functions in PGArray which take the innerType as additional argument? This is basically the same.